### PR TITLE
Added Docker container for Debian

### PR DESCRIPTION
**Pull request by mantoine96**
##### ISSUE TYPE
- Feature Pull Request
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### SUMMARY

Added a Debian 8 Dockerfile for Ansible unit testing / CI. 
I wrote this Dockerfile, adapting from the Ubuntu1404, because sometimes there are things that differ between Ubuntu and Debian, and there could be problems if you are only testing against Ubuntu.

### CODE REVIEW

File: test/utils/docker/debian8/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,84 @@
+FROM debian:jessie
```
Comment (mattclay) [inline test/utils/docker/debian8/Dockerfile]: Use `FROM debian:8` to match the name of the directory.

File: test/utils/docker/debian8/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+RUN apt-get update -y && \
+	DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
```
Comment (mattclay) [inline test/utils/docker/debian8/Dockerfile]: Indent 4 spaces for consistency with the other Dockerfiles.

File: test/utils/docker/debian8/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	curl \
+	debhelper \
+	debianutils \
+	devscripts \
+	docbook-xml \
+	dpkg-dev \
+	fakeroot \
+	gawk \
+	git \
+#	libxml12-utils \
```
Comment (mattclay) [inline test/utils/docker/debian8/Dockerfile]: Remove commented out packages.

File: test/utils/docker/debian8/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+		/etc/init/module*.conf \
+	; do \
+		dpkg-divert --local --rename --add "$f"; \
+	done; \
+	echo '# /lib/init/fstab: cleared out for bare-bones Docker'  > /lib/init/fstab
+
+RUN rm /etc/apt/apt.conf.d/docker-clean
+RUN mkdir /etc/ansible/
+RUN /bin/echo -e "[local]\nlocalhost ansible_connection=local" > /etc/ansible/hosts
+RUN locale-gen en_US.UTF-8
```
Comment (mattclay) [inline test/utils/docker/debian8/Dockerfile]: You'll need to enable this locale first. Adding the following before this line should do the trick:

```
RUN sed -i -e 's/^# \(en_US.UTF-8 UTF-8\)/\1/' /etc/locale.gen
```

File: test/utils/docker/debian8/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+	python-pip \
+	python-setuptools \
+	python-virtualenv \
+	python-yaml \
+	reprepro \
+	rsync \
+	ruby \
+	subversion \
+	sudo \
+	unzip \
```
Comment (mattclay) [inline test/utils/docker/debian8/Dockerfile]: The virtualenv package is needed.

**Comment by mattclay** [comment]
Before merging this Dockerfile it needs to be updated to pass the integration tests.

You can run the tests locally using docker. First you'll need to build your image:

`docker build -t debian8 test/utils/docker/debian8`

Then run the integration tests:

`IMAGE=debian8 test/utils/shippable/integration.sh`

When you encounter an error, you'll need to figure out if it's a problem with the docker image or with the tests and fix accordingly.

**Be sure you've updated your branch to the latest code before testing locally.** There was an issue with the `COPY_SOURCE` option (disabled by default) for `integration.sh` which could cause your source on the host to be deleted on container boot for Debian, as it purges the `/tmp` directory on boot.

**Comment by mantoine96** [comment]
Hi !

First of all thanks for considering my pull request.

I have edited the Dockerfile with your changes, rebased and tried running integration tests. The output seems to match official images' output, so it must be passing (I had difficulties figuring out if the failures were planned failures or true failures)

**Comment by mattclay** [comment]
@thehunt33r Looks like you merged in upstream changes to your branch instead of rebasing, as I see unrelated commits on the PR now.

**Comment by mantoine96** [comment]
@mattclay Oh my bad, I could swear I had rebased. I will try to fix this :)

**Comment by mattclay** [comment]
@thehunt33r I ran the integration tests against this Dockerfile. The tests failed on this task:

```
TASK [test_pip : install the requirement file in a virtualenv] *****************
fatal: [testhost]: FAILED! => {"changed": false, "failed": true, "msg": "Failed to find required executable virtualenv"}

NO MORE HOSTS LEFT *************************************************************
    to retry, use: --limit @destructive.retry

PLAY RECAP *********************************************************************
testhost                   : ok=58   changed=22   unreachable=0    failed=1   

Makefile:117: recipe for target 'destructive' failed
make: *** [destructive] Error 1
```

It looks like the virtualenv package is missing.

**Comment by mattclay** [comment]
Closing this PR due to inactivity.

**Pull request closed without merging**